docs/upgrade.md

```diff
@@ -3,5 +3,263 @@ title: Upgrade guide
+| -------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------- |
+| **ES Modules support**                 | Added ES Modules support via dual CommonJS and ES Module bundling, this enables top-level `await` and tree-shaking.                              | -                    |
+| **Middy.js middleware imports**        | Updated import path for Middy.js middlewares to leverage subpath exports - i.e. `@aws-lambda-powertools/tracer/middleware`.                      | Yes                  |
+| **Types imports**                      | Updated import path for TypeScript types to leverage subpath exports - i.e. `@aws-lambda-powertools/logger/types`.                               | Yes                  |
+| **Logger - log sampling**              | Changed implementation of [log sampling](./core/logger.md#sampling-logs) to dynamically switch log level to `DEBUG` on a percentage of requests. | -                    |
+| **Logger - Custom Log Formatter**      | Updated [custom log formatter](#custom-log-formatter) to include additional persistent attributes along with standard structured keys.           | Yes                  |
+| **Logger & Tracer - helper functions** | Removed deprecated `createLogger` and `createTracer` helper functions in favor of direct instantiation.                                          | Yes                  |
+
+### First steps
+
+Before you start, we suggest making a copy of your current working project or create a new git branch.
+
+1. Upgrade Node.js to v16 or higher, Node.js v20 is recommended.
+2. Ensure that you have the latest Powertools for AWS Lambda (TypeScript) version via [Lambda Layer](./index.md#lambda-layer) or npm.
+3. Review the following sections to confirm whether they apply to your codebase.
+
+## ES Modules support
+
+Starting with v2, Powertools for AWS Lambda (TypeScript) supports ES Modules. This means that you can now import the package using the `import` syntax instead of the `require` syntax. This is especially useful if you want to leverage features like top-level `await` in your Lambda function to run asynchronous code during the initialization phase.
+
+```typescript
+import { getSecret } from '@aws-lambda-powertools/parameters/secrets';
+
+// This code will run during the initialization phase of your Lambda function
+const myApiKey = await getSecret('my-api-key', { transform: 'json' });
+
+export const handler = async () => {
+    // ...
+};
+```
+
+With this change, you can also apply tree-shaking to your function bundle to reduce its size. As part of this release we have made changes to the package and its exports to better support this feature, and we remain committed to improving this in the future based on your feedback.
```

Comment:
```suggestion
In v2, we improved tree-shaking support to help you reduce your function bundle size. We would love to hear your feedback on further improvements we could make.
```

line editing to simplify message and improve cognitive load. Let me know if it's still accurate. 

Add your personal touch (maintainers)

**FROM**

> We remain committed to improving this in the future based on your feedback.

**TO**

> We would love to hear your feedback on further improvements we could make.


quick editing samples to improve in the future:

- `As part of this release` -> `In v2`
- `we have made changes to the package` -> `we changed`